### PR TITLE
apps sc: upgrade harbor to 2.8.2

### LINKED ISSUE #1404: Upgrade harbor to v2.8.1
**What Chart should be upgraded?**
Upgrade harbor to v2.8.1 - https://github.com/goharbor/harbor/releases/tag/v2.8.1

**Why should this Chart be upgraded?**

- [ ] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [x] We need to keep up to prevent future issues.
- [x] Other: *Write the reason here*

lots of enhancements, new features, and bug fixes
**Acceptance criteria**

- I checked the migration of the new Chart:
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

### PR DESCRIPTION
**Pull request by robinAwallace**
**What this PR does / why we need it**:

Upgrades harbor to 2.8.1

**Which issue this PR fixes**: fixes #1404 

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [ ] Proper commit message prefix on all commits
- [x] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [x] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [x] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: WIP-CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ### Release notes
 
+Harbor is upgraded to v2.8.1, which drops support for chartmuseum. If you are using chartmuseum you will need to migrate your [charts](https://github.com/goharbor/harbor/wiki/Migrate-helm-chart-to-oci-registry-in-Harbor).
```
Comment (OlleLarsson) [inline WIP-CHANGELOG.md:3]:
```suggestion
- Harbor is upgraded to v2.8.1, which drops support for chartmuseum. If you are using chartmuseum you will need to migrate your [charts](https://github.com/goharbor/harbor/wiki/Migrate-helm-chart-to-oci-registry-in-Harbor).
```

**Review by OlleLarsson — COMMENTED** [review]
Remember to update the [tests](https://github.com/elastisys/compliantkubernetes-apps/blob/main/pipeline/test/services/service-cluster/) as well :)

**Comment by OlleLarsson** [comment]
Off-topic, but we should add some checks for upgrading charts/apps as there's a lot to remember nowadays
- Update networkpolicies
- Update PSP
- Update tests
- ...

**Comment by robinAwallace** [comment]
Yes, agreed. I will take another look next week to see if i should update anything more.

File: WIP-CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +13,9 @@
 - Added All instances option to OpenSearch dashboard
 - Add networkpolicy generator chart with common rules
 - Added option to enable harbor GC via config
+- Updated harbor chart to `v1.12.1` which upgraded harbor to `v2.8.1`
+  - Dropped support for chartmuseum
+  - Default login page is now OIDC, for local login refer to path `<url>/account/sign-in`
```
Comment (viktor-f) [inline WIP-CHANGELOG.md:18]: Nit: shouldn't this be under `changed` or `updated` instead of `added`?

Comment (robinAwallace) [reply WIP-CHANGELOG.md]: Fixed

File: WIP-CHANGELOG.md [hunk 3]
```diff
@@ -29,3 +32,6 @@
 - Update Trivy Operator Dashboard to improve the user experience
 - Bump sops to version `3.7.3`.
 - Update Falco rules
+- Updated harbor chart to `v1.12.1` which upgraded harbor to `v2.8.1`
```
Comment (viktor-f) [inline WIP-CHANGELOG.md:35]: They have now released chart version 1.12.2 with harbor version 2.8.2. So if you want you could try to upgrade to that, seemed like there were only some small changes.

Comment (robinAwallace) [reply WIP-CHANGELOG.md:35]: Right, forgot about that. Will do that :+1:

Comment (robinAwallace) [reply WIP-CHANGELOG.md]: Done

**Comment by robinAwallace** [comment]
@viktor-f @davidumea @Ajarmar Can you take a look at this PR?

**Review by OlleLarsson — APPROVED** [review]
LGTM, maybe if there's time, implement the networkpolicies via the new netpol generator chart?

File: WIP-CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,8 @@
 ### Release notes
 
+- Harbor is upgraded to v2.8.1, which drops support for chartmuseum. If you are using chartmuseum you will need to migrate your [charts](https://github.com/goharbor/harbor/wiki/Migrate-helm-chart-to-oci-registry-in-Harbor).
```
Comment (davidumea) [inline WIP-CHANGELOG.md:3]: slight nit
```suggestion
- Harbor is upgraded to v2.8.2, which drops support for chartmuseum. If you are using chartmuseum you will need to migrate your [charts](https://github.com/goharbor/harbor/wiki/Migrate-helm-chart-to-oci-registry-in-Harbor).
```

**Review by davidumea — APPROVED** [review]

**Review by viktor-f — APPROVED** [review]

**Comment by robinAwallace** [comment]
> LGTM, maybe if there's time, implement the networkpolicies via the new netpol generator chart?

Yes i can take a look at the netpols

**Comment by robinAwallace** [comment]
I will merge this and open a new PR for np.

**Pull request merged**